src/internal/storage/fileset/option.go

```diff
@@ -55,6 +55,15 @@ func WithRenewal(ttl time.Duration, r *renew.StringSet) UnorderedWriterOption {
 	}
 }
 
+// WithParentID sets the parent fileset ID for the unordered writer.
+// This is used for converting directory deletions into a set of point
+// deletes for the files contained within the directory.
+func WithParentID(parentID *ID) UnorderedWriterOption {
```

Comment:
Shouldn't this be an option on the Writer?

Reply:
I think it makes more sense for the unordered writer. The writer has more restrictions on the ordering of paths, so it is going to be cleaner to have it operate on the fully resolved paths.